### PR TITLE
Convert kubectl kubemark probe files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/probe/http/http.go [hunk 1]
```diff
@@ -120,21 +120,21 @@ func DoHTTPProbe(url *url.URL, headers http.Header, client GetHTTPInterface) (pr
...
+			klog.V(4).InfoS("Non fatal body truncation for URL", "url", url.String(), "res", *res)
 		} else {
 			return probe.Failure, "", err
 		}
 	}
 	body := string(b)
 	if res.StatusCode >= http.StatusOK && res.StatusCode < http.StatusBadRequest {
 		if res.StatusCode >= http.StatusMultipleChoices { // Redirect
-			klog.V(4).Infof("Probe terminated redirects for %s, Response: %v", url.String(), *res)
+			klog.V(4).InfoS("Probe terminated redirects for URL", url.String(), *res)
```
Comment (jsturtevant) [inline pkg/probe/http/http.go:131]:
```suggestion
			klog.V(4).InfoS("Probe terminated redirects for URL", "url", url.String(), "response", *res)
```

for consistency with other logs

File: pkg/probe/http/http.go [hunk 2]
```diff
@@ -120,21 +120,21 @@ func DoHTTPProbe(url *url.URL, headers http.Header, client GetHTTPInterface) (pr
 	b, err := utilio.ReadAtMost(res.Body, maxRespBodyLength)
 	if err != nil {
 		if err == utilio.ErrLimitReached {
-			klog.V(4).Infof("Non fatal body truncation for %s, Response: %v", url.String(), *res)
+			klog.V(4).InfoS("Non fatal body truncation for URL", "url", url.String(), "res", *res)
```
Comment (jsturtevant) [inline pkg/probe/http/http.go:123]:
```suggestion
			klog.V(4).InfoS("Non fatal body truncation for URL", "url", url.String(), "response", *res)
```

File: pkg/kubemark/hollow_kubelet.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ type HollowKubletOptions struct {
 func GetHollowKubeletConfig(opt *HollowKubletOptions) (*options.KubeletFlags, *kubeletconfig.KubeletConfiguration) {
 	testRootDir := utils.MakeTempDirOrDie("hollow-kubelet.", "")
 	podFilePath := utils.MakeTempDirOrDie("static-pods", testRootDir)
-	klog.Infof("Using %s as root dir for hollow-kubelet", testRootDir)
+	klog.InfoS("Root dir for hollow-kubelet", "testRootDir", testRootDir)
```
Comment (jsturtevant) [inline pkg/kubemark/hollow_kubelet.go:156]:
```suggestion
	klog.InfoS("Root dir for hollow-kubelet", "rootDir", testRootDir)
```

**Review by jsturtevant — COMMENTED** [review]
/ok-to-test

**Comment by kikisdeliveryservice** [comment]
related-to: https://github.com/kubernetes/enhancements/issues/1602

File: pkg/kubemark/hollow_kubelet.go (hunk 3 above)
Comment (scottilee) [reply pkg/kubemark/hollow_kubelet.go:156]: Fixed.

File: pkg/probe/http/http.go (hunk 2 above)
Comment (scottilee) [reply pkg/probe/http/http.go:123]: Fixed.

File: pkg/probe/http/http.go (hunk 1 above)
Comment (scottilee) [reply pkg/probe/http/http.go]: Fixed.

**Comment by scottilee** [comment]
@jsturtevant Thanks for the review! I addressed your comments and updated the .structured_logging file.

**Comment by scottilee** [comment]
Fixed "pull-kubernetes-verify" errors.

**Review by ehashman — COMMENTED** [review]
/sig scalability
for kubemark approval

/wg structured-logging

File: pkg/kubectl/cmd/convert/convert.go [hunk 4]
```diff
@@ -221,7 +221,7 @@ func asVersionedObject(infos []*resource.Info, forceList bool, specifiedOutputVe
 		if len(actualVersion.Version) > 0 {
 			defaultVersionInfo = fmt.Sprintf("Defaulting to %q", actualVersion.Version)
 		}
-		klog.V(1).Infof("info: the output version specified is invalid. %s\n", defaultVersionInfo)
+		klog.V(1).InfoS("Info: the output version specified is invalid.\n", "defaultVersionInfo", defaultVersionInfo)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/kubectl/cmd/convert/convert.go:224]: Fixed.

File: pkg/kubemark/controller.go [hunk 5]
```diff
@@ -339,7 +341,7 @@ func (kubemarkController *KubemarkController) runNodeCreation(stop <-chan struct
 			kubemarkController.nodeGroupQueueSizeLock.Lock()
 			err := kubemarkController.addNodeToNodeGroup(nodeGroup)
 			if err != nil {
-				klog.Errorf("failed to add node to node group %s: %v", nodeGroup, err)
+				klog.ErrorS(err, "Failed to add node to node group", "nodeGroup", nodeGroup)
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/kubemark/controller.go:344]: @ehashman How do I reference the node here?

File: pkg/probe/exec/exec.go [hunk 6]
```diff
@@ -75,7 +75,7 @@ func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
 				return probe.Failure, string(data), nil
 			}
 
-			klog.Warningf("Exec probe timed out after %s but ExecProbeTimeout feature gate was disabled", timeoutErr.Timeout())
+			klog.InfoS("Exec probe timed out after %s but ExecProbeTimeout feature gate was disabled", "timeout", timeoutErr.Timeout())
```
[the remark this answers is not in the document]
Comment (scottilee) [reply pkg/probe/exec/exec.go:78]: Fixed.

File: pkg/kubemark/controller.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (ehashman) [reply pkg/kubemark/controller.go:344]: Looks like it's not returned by `addNodeToNodeGroup`, so this is fine

**Review by ehashman — COMMENTED** [review]
/lgtm

File: pkg/kubemark/controller.go [hunk 7]
```diff
@@ -251,7 +253,7 @@ func (kubemarkController *KubemarkController) RemoveNodeFromNodeGroup(nodeGroup
 		err = kubemarkController.externalCluster.client.CoreV1().ReplicationControllers(namespaceKubemark).Delete(context.TODO(), pod.ObjectMeta.Labels["name"],
 			metav1.DeleteOptions{PropagationPolicy: &policy})
 		if err == nil {
-			klog.Infof("marking node %s for deletion", node)
+			klog.InfoS("Marking Node for deletion", "node", node)
```
Comment (mrunalp) [inline pkg/kubemark/controller.go:256]: nit: `Node` -> `node`

**Comment by ehashman** [comment]
/lgtm cancel
while waiting on change

File: pkg/kubemark/controller.go (hunk 7 above)
Comment (pacoxu) [reply pkg/kubemark/controller.go:256]: Only this small nit.
Would you update the PR?

Comment (scottilee) [reply pkg/kubemark/controller.go]: Fixed.

**Comment by scottilee** [comment]
@pacoxu @ehashman @mrunalp Fixed the nit.

**Review by ehashman — COMMENTED** [review]
/lgtm

**Comment by pacoxu** [comment]
/lgtm
/assign @derekwaynecarr @pwittrock @shyamjvs
for approval

**Review by mrunalp — APPROVED** [review]

**Pull request closed without merging**